### ISSUE TITLE
Circular progress indicators are not accessible

### ISSUE THREAD

**Issue by steverep** [home-assistant/frontend#14177]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

The `ha-circular-progress` element is nearly always used inaccessibly, mostly because of the reasons in material-components/material-web#3236.  In summary:

- **Accessible Name:** An `alt` attribute is used in some cases, but this does nothing.  The component supports `aria-label`, but it's not documented in the API.  Note though that this should label what the progress indicates, e.g. "Installation", and not its current state, e.g. "Loading".
- **Progress State:** Without getting into the details, either a valid `progress` attribute or the `indeterminate` attribute must be present, or the ARIA markup is invalid.
- **Not valid inside buttons:** In some cases, the element is used inside an `mwc-button`, which is not valid because the children of a button role are treated as for presentation only.  Also, this is not used consistently throughout the UI and IMHO should just be eliminated or replaced with a simple icon with a text alternative per #14170.

### Describe the behavior you expected

Elements with the `progressbar` role should have an accessible name and state, and should not be inside buttons.

### Steps to reproduce the issue

n/a

### What version of Home Assistant Core has the issue?

2022.10.x

### In which browser are you experiencing the issue with?

n/a

### Which operating system are you using to run this browser?

n/a

**Comment by steverep** [comment]
Still an issue

**Comment by steverep** [comment]
Still an issue

**Comment by silamon** [comment]
The [material m3 documentation](https://m3.material.io/components/progress-indicators/accessibility#208535fe-f0ae-49d6-85fa-ecb4a253fd88) mentions that it is allowed to integrate progress components into a button. 

I've set up a [playground](https://lit.dev/playground/#project=W3sibmFtZSI6ImltcG9ydHMudHMiLCJjb250ZW50IjoiaW1wb3J0IFwiQG1hdGVyaWFsL3dlYi9hbGxcIjtcblxuIn0seyJuYW1lIjoiaW5kZXguaHRtbCIsImNvbnRlbnQiOiI8IURPQ1RZUEUgaHRtbD5cbjxoZWFkPlxuICA8c2NyaXB0IHR5cGU9XCJtb2R1bGVcIiBzcmM9XCIuL2ltcG9ydHMuanNcIj48L3NjcmlwdD5cbjwvaGVhZD5cbjxib2R5PlxuICA8bWQtY2lyY3VsYXItcHJvZ3Jlc3MgaW5kZXRlcm1pbmF0ZSBhcmlhLWxhYmVsPVwiQmFzaWMgY2lyY3VsYXItcHJvZ3Jlc3NcIj5cbiAgPC9tZC1jaXJjdWxhci1wcm9ncmVzcz5cbiAgXG4gIDxtZC1vdXRsaW5lZC1idXR0b24-XG4gICAgICA8bWQtY2lyY3VsYXItcHJvZ3Jlc3MgaW5kZXRlcm1pbmF0ZSBhcmlhLWxhYmVsPVwiQmFzaWMgY2lyY3VsYXItcHJvZ3Jlc3MgaW4gYSBidXR0b25cIj5cbiAgICAgIDwvbWQtY2lyY3VsYXItcHJvZ3Jlc3M-XG4gIDwvbWQtb3V0bGluZWQtYnV0dG9uPlxuIFxuXG48L2JvZHk-XG4ifSx7Im5hbWUiOiJwYWNrYWdlLmpzb24iLCJjb250ZW50Ijoie1xuICBcImRlcGVuZGVuY2llc1wiOiB7XG4gICAgXCJsaXRcIjogXCJeMy4wLjAtcHJlLjFcIixcbiAgICBcIkBsaXQvcmVhY3RpdmUtZWxlbWVudFwiOiBcIl4yLjAuMC1wcmUuMVwiLFxuICAgIFwibGl0LWVsZW1lbnRcIjogXCJeNC4wLjAtcHJlLjFcIixcbiAgICBcImxpdC1odG1sXCI6IFwiXjMuMC4wLXByZS4xXCJcbiAgfVxufSIsImhpZGRlbiI6dHJ1ZX1d) with a circular progress in a button. Can you check if that is an accessible example?

**Comment by steverep** [comment]
> The [material m3 documentation](https://m3.material.io/components/progress-indicators/accessibility#208535fe-f0ae-49d6-85fa-ecb4a253fd88) mentions that it is allowed to integrate progress components into a button.

What can I say, that's bad advice.  I can open an issue with them on it.

> I've set up a [playground](https://lit.dev/playground/#project=W3sibmFtZSI6ImltcG9ydHMudHMiLCJjb250ZW50IjoiaW1wb3J0IFwiQG1hdGVyaWFsL3dlYi9hbGxcIjtcblxuIn0seyJuYW1lIjoiaW5kZXguaHRtbCIsImNvbnRlbnQiOiI8IURPQ1RZUEUgaHRtbD5cbjxoZWFkPlxuICA8c2NyaXB0IHR5cGU9XCJtb2R1bGVcIiBzcmM9XCIuL2ltcG9ydHMuanNcIj48L3NjcmlwdD5cbjwvaGVhZD5cbjxib2R5PlxuICA8bWQtY2lyY3VsYXItcHJvZ3Jlc3MgaW5kZXRlcm1pbmF0ZSBhcmlhLWxhYmVsPVwiQmFzaWMgY2lyY3VsYXItcHJvZ3Jlc3NcIj5cbiAgPC9tZC1jaXJjdWxhci1wcm9ncmVzcz5cbiAgXG4gIDxtZC1vdXRsaW5lZC1idXR0b24-XG4gICAgICA8bWQtY2lyY3VsYXItcHJvZ3Jlc3MgaW5kZXRlcm1pbmF0ZSBhcmlhLWxhYmVsPVwiQmFzaWMgY2lyY3VsYXItcHJvZ3Jlc3MgaW4gYSBidXR0b25cIj5cbiAgICAgIDwvbWQtY2lyY3VsYXItcHJvZ3Jlc3M-XG4gIDwvbWQtb3V0bGluZWQtYnV0dG9uPlxuIFxuXG48L2JvZHk-XG4ifSx7Im5hbWUiOiJwYWNrYWdlLmpzb24iLCJjb250ZW50Ijoie1xuICBcImRlcGVuZGVuY2llc1wiOiB7XG4gICAgXCJsaXRcIjogXCJeMy4wLjAtcHJlLjFcIixcbiAgICBcIkBsaXQvcmVhY3RpdmUtZWxlbWVudFwiOiBcIl4yLjAuMC1wcmUuMVwiLFxuICAgIFwibGl0LWVsZW1lbnRcIjogXCJeNC4wLjAtcHJlLjFcIixcbiAgICBcImxpdC1odG1sXCI6IFwiXjMuMC4wLXByZS4xXCJcbiAgfVxufSIsImhpZGRlbiI6dHJ1ZX1d) with a circular progress in a button. Can you check if that is an accessible example?

Nope.

**Comment by silamon** [comment]
Too bad, thanks for checking. I will open a pull request to solve the first two bullets but  it would have been nice if the third was solved as well. Now putting some work in m3, we should take accessibility in mind. The button with circulair progress spinner is a visual indication that  for example saving is happening. Icon buttons or the svg alternative you have mentioned are not a good alternative to provide the same experience as now.
